src/main/java/seedu/address/logic/commands/EmailSpecificCommand.java

```diff
@@ -0,0 +1,189 @@
+
+import javax.mail.Authenticator;
+import javax.mail.PasswordAuthentication;
+import javax.mail.Session;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.logic.CommandHistory;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.person.Person;
+import seedu.address.model.tag.Tag;
+
+//@@author aaryamNUS
+/**
+ * Sends an email to any guests that have at least one of the specified tags
+ */
+public class EmailSpecificCommand extends Email {
+
+    public static final String COMMAND_WORD = "emailSpecific";
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Sends an email to the guests that have "
+            + "at least one of the provided Tags.\n"
+            + "Parameters: "
+            + "[" + PREFIX_TAG + "TAG]...\n"
+            + "Example: " + COMMAND_WORD + " "
+            + PREFIX_TAG + "VIP " + PREFIX_TAG + "Paid";
+
+    private static final String MESSAGE_MAIL_PERSON_SUCCESS = "Successfully sent an email to %1$d emails, "
+            + "did not send an email to %2$d guests!";
+
+    private static Logger logger = Logger.getLogger("execute");
+    private static String username;
+    private static String password;
+    private final Set<Tag> tagsToSend;
+
+    /**
+     * @param tagsToSend of the guests with the specified tags
+     */
+    public EmailSpecificCommand(Set<Tag> tagsToSend) {
+        requireNonNull(tagsToSend);
+        this.tagsToSend = tagsToSend;
+    }
+
+    /**
+     * Sends an email to all guests that have at least one of the specified tags
+     * @param model must be non-null
+     */
+    @Override
+    public CommandResult execute(Model model, CommandHistory history) throws CommandException {
+        requireNonNull(model);
+        List<Person> lastShownList = model.getFilteredPersonList();
+        String emailSubject;
+        String emailMessage;
+
+        // Check for duplicate emails to ensure each guest only receives one email, even if
+        // multiple guests have registered under the same email
+        HashSet<String> personsToSendEmail = new HashSet<>();
+
+        // Used to build the final recipients String
+        StringBuilder personsToEmail = new StringBuilder();
+        String recipients;
+
+        // First check which of the guests have at least one of the specified tags,
+        // and filter out any duplicate guests
+        for (Person individualGuests : lastShownList) {
+            for (Tag singleTags : tagsToSend) {
+                if (individualGuests.getTags().contains(singleTags)) {
+                    if (personsToSendEmail.contains(individualGuests.getEmail().toString())) {
+                        logger.log(Level.INFO, "Guest email address has already been sent an email!");
+                    } else {
+                        personsToSendEmail.add(individualGuests.getEmail().toString());
+                    }
+                }
+            }
```

Comment:
Might want to consider doing SLAP on this and abstract out this chunk to avoid deeply nested code